attestation-agent/attester/src/lib.rs

```diff
@@ -77,6 +82,11 @@ pub fn detect_tee_type() -> Option<Tee> {
         return Some(Tee::Sgx);
     }
 
+    #[cfg(feature = "az-tdx-vtpm-attester")]
+    if az_tdx_vtpm::detect_platform() {
+        return Some(Tee::AzTdxVtpm);
+    }
```

Comment:
Note: the order is critical here, because AzSnpVtpm has a weak detection heuristic and will report positive if it's able to pull an HCL report, even if it's a tdx one. We need to provide a better heuristic on az-snp-vtpm to detect azure snp cvms,